code/game/atoms_movable.dm

```diff
@@ -272,3 +275,17 @@
 	if (src.master)
 		return src.master.attack_hand(a, b, c)
 	return
+
+/atom/movable/proc/handle_buckled_mob_movement(loc,direct)
+	if(!buckled_mob.Move(loc, direct))
+		loc = buckled_mob.loc
+		last_move = buckled_mob.last_move
+		inertia_dir = last_move
+		buckled_mob.inertia_dir = last_move
+		. = 0

```

Comment:
you need a return after that, otherwise it will call ". = 1" and always return 1. Or maybe just use "return 0" and "return 1".